cloud/amazon/ec2_vpc_peer.py

```diff
@@ -0,0 +1,312 @@
+    state: accept
+
+
+# Complete example to create and reject a cross account peering connection.
+- name: Create cross account VPC peering Connection
+  ec2_vpc_peer:
+    region: ap-southeast-2
+    vpc_id: vpc-12345678
+    peer_vpc_id: vpc-12345678
+    peer_owner_id: 123456789102
+    state: present
+  register: vpc_peer
+
+- name: Reject a cross account VPC peering Connection
+  ec2_vpc_peer:
+    region: ap-southeast-2
+    peering_id: "{{ vpc_peer.peering_id }}"
+    profile: bot03_profile_for_cross_account
+    state: reject
+
+'''
+RETURN = '''
+task:
+  description: details about the tast that was started
+  type: complex
+  sample: "TODO: include sample"
+'''
+
+try:
+    import json
+    import datetime
+    import boto
+    import botocore
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+    
+
+def describe_peering_connections(vpc_id, peer_vpc_id, client):
+    result = client.describe_vpc_peering_connections(Filters=[
+        {'Name': 'requester-vpc-info.vpc-id', 'Values': [vpc_id]},
+        {'Name': 'accepter-vpc-info.vpc-id', 'Values': [peer_vpc_id]}
+        ])
+    if result['VpcPeeringConnections'] == []:
+        result = client.describe_vpc_peering_connections(Filters=[
+            {'Name': 'requester-vpc-info.vpc-id', 'Values': [peer_vpc_id]},
+            {'Name': 'accepter-vpc-info.vpc-id', 'Values': [vpc_id]}
+            ])
+    return result
+
+
+def is_active(peering_conn):
+    return peering_conn['Status']['Code'] == 'active'
+
+
+def is_pending(peering_conn):
+    return peering_conn['Status']['Code'] == 'pending-acceptance'
+
+
+def peer_status(resource, module):
+    peer_id = module.params.get('peering_id')
+    vpc_peering_connection = resource.VpcPeeringConnection(peer_id)
+    return vpc_peering_connection.status['Message']
+
+
+def create_peer_connection(client, module):
+    changed = False
+    vpc_id = module.params.get('vpc_id')
+    peer_vpc_id = module.params.get('peer_vpc_id')
+    peer_owner_id = module.params.get('peer_owner_id', False)
+    peering_conns = describe_peering_connections(vpc_id, peer_vpc_id, client)
+    for peering_conn in peering_conns['VpcPeeringConnections']:
+        if is_active(peering_conn):
+            return (False, peering_conn['VpcPeeringConnectionId'])
+        if is_pending(peering_conn):
+            return (False, peering_conn['VpcPeeringConnectionId'])
+    if not peer_owner_id:
+        try:
+            peering_conn = client.create_vpc_peering_connection(VpcId=vpc_id, PeerVpcId=peer_vpc_id)
+            return (True, peering_conn['VpcPeeringConnection']['VpcPeeringConnectionId'])
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=str(e))            
+    else:
+        try:
+            peering_conn = client.create_vpc_peering_connection(VpcId=vpc_id, PeerVpcId=peer_vpc_id, PeerOwnerId=str(peer_owner_id))
```

Comment:
Pretty long line here - would it be better to declare a params = dict() and then set the params options first and then do peering_conn = client.create_vpc_peering_connection(**params). That way you can also ensure you have the info from the user required?